java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -0,0 +1,656 @@
+        currentBlock = createUnconditionalJump(throwStatementTree, exitBlock);
+        build(throwStatementTree.expression());
+        break;
+      }
+      case SYNCHRONIZED_STATEMENT: {
+        SynchronizedStatementTree sst = (SynchronizedStatementTree) tree;
+        //Naively build synchronized statement.
+        build(sst.block());
+        build(sst.expression());
+        break;
+      }
+      case POSTFIX_INCREMENT:
+      case POSTFIX_DECREMENT:
+      case PREFIX_INCREMENT:
+      case PREFIX_DECREMENT:
+      case UNARY_MINUS:
+      case UNARY_PLUS:
+      case BITWISE_COMPLEMENT:
+      case LOGICAL_COMPLEMENT:
+        UnaryExpressionTree e = (UnaryExpressionTree) tree;
+        currentBlock.elements.add(e);
+        build(e.expression());
+        break;
+      case PARENTHESIZED_EXPRESSION:
+        build(((ParenthesizedTree) tree).expression());
+        break;
+      case ARRAY_ACCESS_EXPRESSION: {
+        ArrayAccessExpressionTree aaet = (ArrayAccessExpressionTree) tree;
+        currentBlock.elements.add(aaet);
+        build(aaet.expression());
+        build(aaet.dimension());
+        break;
+      }
+      case ARRAY_DIMENSION: {
+        ArrayDimensionTree arrayDimensionTree = (ArrayDimensionTree) tree;
+        if(arrayDimensionTree.expression() != null) {
+          build(arrayDimensionTree.expression());
+        }
+        break;
+      }
+      case NEW_CLASS: {
+        NewClassTree newClassTree = (NewClassTree) tree;
+        currentBlock.elements.add(newClassTree);
+        if(newClassTree.enclosingExpression() != null) {
+          build(newClassTree.enclosingExpression());
+        }
+        build(Lists.reverse(newClassTree.arguments()));
+      }
+      break;
+      case IDENTIFIER:
+      case INT_LITERAL:
+      case LONG_LITERAL:
+      case DOUBLE_LITERAL:
+      case CHAR_LITERAL:
+      case FLOAT_LITERAL:
+      case STRING_LITERAL:
+      case BOOLEAN_LITERAL:
+      case NULL_LITERAL:
+        currentBlock.elements.add(tree);
+        break;
+      case TYPE_CAST:
+        currentBlock.elements.add(tree);
+        TypeCastTree typeCastTree = (TypeCastTree) tree;
+        build(typeCastTree.expression());
+        break;
+      case INSTANCE_OF:
+        InstanceOfTree instanceOfTree = (InstanceOfTree) tree;
+        currentBlock.elements.add(instanceOfTree);
+        build(instanceOfTree.expression());
+        break;
+      case EMPTY_STATEMENT:
+        break;
+      case NEW_ARRAY:
+        NewArrayTree newArrayTree = (NewArrayTree) tree;
+        currentBlock.elements.add(tree);
+        build(Lists.reverse(newArrayTree.dimensions()));
+        build(Lists.reverse(newArrayTree.initializers()));
+        break;
+      case CLASS:
+      case ENUM:
+      case ANNOTATION_TYPE:
+      case INTERFACE:
+        //Ignore declarations
+        break;
+      //Java 8 constructions : ignored for now.
+      case METHOD_REFERENCE:
+      case LAMBDA_EXPRESSION:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This case's code block is the same as the block for the case on line 541. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1871)